### DISCUSSION TITLE
latest shortcode added

### DISCUSSION THREAD

**Discussion by f14m07** [liedekef/events-made-easy#609] [General]
Hi ,
The latest shortcode added does the same thing as EMEFS. Are the two systems working well together at the moment? I imagine that the long-term aim is to abandon EMEFS?

**Comment by liedekef** [comment]
Indeed it does the same but with less code duplication and more easily maintainable. EMEFS just got an upgrade too to warn about it going to be discontinued in the future. Both shortcodes work fine for now.

**Reply by f14m07** [reply]
With EMEFS, the first 2 letters of the location were used to display a list of locations beginning with these two letters. All you had to do was choose. This no longer seems to be the case. Is this still possible?

**Reply by liedekef** [reply]
This is still the case and working as before ...

**Reply by f14m07** [reply]
On my local site I have a page with the short code EMEFS and the two letters display the list. But on the page with the new shortcode, it doesn't work, the two letters display nothing. I use the Default form format and the "Force location creation?" is on "No".
Have I forgotten something?

**Reply by liedekef** [reply]
It should work. I tested it on a public site and on my local site: both ok. Can you demo it?

**Reply by f14m07** [reply]
This is what I have in the console
<img width="740" alt="Capture d’écran 2024-07-26 à 17 29 16" src="https://github.com/user-attachments/assets/58add12e-7024-4a25-9fdd-45ce8cdb65e3">

**Reply by liedekef** [reply]
well line 86 n eme-fs.js is " emefs.translate_map_is_active". Now wordpress translates (=replaces) "emefs." followed by strings when I load that javascript file. So it is weird that your site shows that ... the translation happens in eme-fs.php, line 42-48:
```
        $translation_array = [ 
                'translate_ajax_url' => admin_url( 'admin-ajax.php' ),
                'translate_map_is_active' => $map_enabled,
                'translate_map_zooming' => $zooming_enabled,
                'translate_frontendnonce' => wp_create_nonce( 'eme_frontend' )
        ];
        wp_localize_script( 'eme-fs-location', 'emefs', $translation_array );
        wp_enqueue_script( 'eme-fs-location' );
```
So something on your site is blocking this somehow??? I see litespeed active , so I'm guessing it is a caching issue :-)

**Reply by f14m07** [reply]
I have the twice one page with the new shortcode and one with shortcode EMEFS

**Reply by f14m07** [reply]
It is the same issue on local and production site. On the local site ther is no litespeed

**Reply by f14m07** [reply]
It is weird that the page with the shortcode EMEFS works well but not the other one

**Reply by liedekef** [reply]
I have both active: both work fine ... what happens if you deactivate emefs?

**Reply by liedekef** [reply]
Demo of both together: https://www.e-dynamics.be/wordpress/emefs-new/ (new) and https://www.e-dynamics.be/wordpress/account/ (old)

**Reply by f14m07** [reply]
desactive emefs change nothing

**Reply by f14m07** [reply]
I can't access to emefs-new

**Reply by f14m07** [reply]
I will looking for giving you access to demo

**Reply by liedekef** [reply]
then I would need a demo (no screenshot) without caching. As you can see in my demo-links are work fine. When looking at the source code of the page where the new shortcode is, you should see something like this:

```
<script id="eme-fs-location-js-extra">
var emefs = {"translate_ajax_url":"http:\/\/localhost\/wordpress\/wp-admin\/admin-ajax.php","translate_map_is_active":"true","translate_map_zooming":"true","translate_frontendnonce":"6019eb9aae"};
</script>
<script src="http://localhost/wordpress/wp-content/plugins/events-made-easy/js/eme_fs.js?ver=2.5.0" id="eme-fs-location-js"></script>
```
making emefs a variable that can then be used in eme_fs.js (that's the way wordpress works)

**Reply by liedekef** [reply]
> I can't access to emefs-new

wait ... maybe that's the issue: guest submit. That might be it, I'll check here. Are you logged in as admin when doing the test?
Nope, works fine here too

**Reply by f14m07** [reply]
You should access to my local site with this URL exclusive-sentence.localsite.io
username 'xxx' pwd 'xxx'

**Reply by liedekef** [reply]
ok, I'm in, where is the page for a new event?

**Reply by f14m07** [reply]
I give you the link to new page : /proposer-une-nouvelle-navigation/
and with emefs : /proposer-une-sortie/

**Comment by f14m07** [comment]
When looking at the source code of the page where the new shortcode is, I don't see what you said, no var emefs=

**Reply by liedekef** [reply]
I can't test it, since I need to be logged in to test  :-) Can you open guest submit? I already found that the message for "guest not allowed" was not returned correctly, but that's a small issue.

**Comment by f14m07** [comment]
You can login the site with the same login password i gave you

**Reply by liedekef** [reply]
ow ... I deleted that info here (too public) ... can you mail that to me again?)

**Comment by liedekef** [comment]
Ok, I think I fixed it on your site, see this changeset:
https://github.com/liedekef/events-made-easy/commit/477fa0847ee6fbc51c7039b77c8e65181b6d85cd

Test it :-)

**Reply by f14m07** [reply]
Yes ! It works now on my local site.

**Comment by f14m07** [comment]
Thanks a lot for your work. But what was the cause. The plugin EMEFS? or something else. And why it worked on your demo site

**Reply by liedekef** [reply]
I really don't know why it works in my setup ... but: the script localise part didn't happen on your end for the new javascript, so I checked where I did the localising for the "normal" eme javascript file and that happened earlier (during a wp hook). So I now moved that code over to that section and now it works. In short: the localising now happens earlier but the reason why it worked for me must be a combo of theme and good fortune :-)

**Comment by liedekef** [comment]
btw, now that it works: if you have any more issues let me know!

**Reply by f14m07** [reply]
I don't have time to migrate at the moment, but I'll be sure to come back if I have any problems.
